diff --git a/cmake/Modules/FindTinyXML2.cmake b/cmake/Modules/FindTinyXML2.cmake
index 87991be..ddf647d 100644
--- a/cmake/Modules/FindTinyXML2.cmake
+++ b/cmake/Modules/FindTinyXML2.cmake
@@ -76,3 +76,13 @@ endif()
 if(NOT TinyXML2_LIBRARIES)
   set(TinyXML2_LIBRARIES ${TINYXML2_LIBRARY})
 endif()
+
+# On case-insensitive filesystem, it is possible that FindTinyXML2.cmake is used if the caller
+# invoked find_package(TINYXML2), that is the signature used by gz-cmake, see
+# https://github.com/gazebosim/gz-cmake/blob/gz-cmake4_4.1.0/cmake/FindTINYXML2.cmake
+# If that is the case (and we detect it by checking the value of CMAKE_FIND_PACKAGE_NAME)
+# we also define a TINYXML2::TINYXML2 target for gz-cmake compatibility
+if(TARGET tinyxml2::tinyxml2 AND CMAKE_FIND_PACKAGE_NAME STREQUAL "TINYXML2" AND NOT TARGET TINYXML2::TINYXML2)
+  add_library(TINYXML2::TINYXML2 INTERFACE IMPORTED)
+  set_property(TARGET tinyxml2::tinyxml2 PROPERTY INTERFACE_LINK_LIBRARIES tinyxml2::tinyxml2)
+endif()
