diff --git a/CMakeLists.txt b/CMakeLists.txt
index e07dcf4..8d23997 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,13 +23,21 @@ find_package(pcl_conversions REQUIRED)
 find_package(tf2_eigen REQUIRED)
 
 # ==== Options ====
-add_compile_options(-Wall -Wextra)
+if(MSVC)
+  add_compile_options(/W2)
+  add_compile_definitions(NOMINMAX _USE_MATH_DEFINES WIN32_LEAN_AND_MEAN)
+else()
+  add_compile_options(-Wall -Wextra)
+endif()
+
 if(NOT DEFINED CMAKE_CXX_STANDARD)
   set(CMAKE_CXX_STANDARD 17)
   set(CMAKE_CXX_STANDARD_REQUIRED ON)
 endif()
 option(CMAKE_POSITION_INDEPENDENT_CODE "Build position independent code." ON)
 
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+
 set(_ouster_ros_INCLUDE_DIRS
   include
   ouster-sdk/ouster_client/include
