diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5b54eb0b4..18150e6a5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,9 +9,12 @@ option(
   OFF
 )
 
-add_compile_options(-Wall)
-add_compile_options(-Wextra)
-add_compile_options(-Wno-unused-parameter)
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall)
+  add_compile_options(-Wextra)
+  add_compile_options(-Wno-unused-parameter)
+endif()
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
 
 if(NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
   message("${PROJECT_NAME}: You did not request a specific build type: selecting 'RelWithDebInfo'.")
diff --git a/include/ur_robot_driver/hardware_interface.hpp b/include/ur_robot_driver/hardware_interface.hpp
index e9ea08129..8eba2be91 100644
--- a/include/ur_robot_driver/hardware_interface.hpp
+++ b/include/ur_robot_driver/hardware_interface.hpp
@@ -64,6 +64,8 @@
 #include "geometry_msgs/msg/transform_stamped.hpp"
 #include "tf2_geometry_msgs/tf2_geometry_msgs.hpp"
 
+typedef unsigned int uint;
+
 namespace ur_robot_driver
 {
 enum class PausingState
diff --git a/src/hardware_interface.cpp b/src/hardware_interface.cpp
index 5e2e5730d..51d093547 100644
--- a/src/hardware_interface.cpp
+++ b/src/hardware_interface.cpp
@@ -42,6 +42,7 @@
 #include <utility>
 #include <vector>
 
+#include "ur_robot_driver/hardware_interface.hpp"
 #include "ur_client_library/exceptions.h"
 #include "ur_client_library/ur/tool_communication.h"
 #include "ur_client_library/ur/version_information.h"
diff --git a/src/robot_state_helper.cpp b/src/robot_state_helper.cpp
index c422ae951..0a57ae75d 100755
--- a/src/robot_state_helper.cpp
+++ b/src/robot_state_helper.cpp
@@ -366,7 +366,7 @@ void RobotStateHelper::setModeExecute(const std::shared_ptr<RobotStateHelper::Se
           result_->message = "Play program service not available on this robot.";
         } else {
           // The dashboard denies playing immediately after switching the mode to RUNNING
-          sleep(1);
+          std::this_thread::sleep_for(std::chrono::milliseconds(1000));
           result_->success = safeDashboardTrigger(this->play_program_srv_);
         }
       }
