diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7ba08db..9783c15 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,7 +16,11 @@
 cmake_minimum_required(VERSION 3.16)
 project(ffmpeg_encoder_decoder)
 
-add_compile_options(-Wall -Wextra -Wpedantic -Werror)
+if(MSVC)
+  add_compile_options(/W4)
+else()
+  add_compile_options(-Wall -Wextra -Wpedantic -Werror)
+endif()
 
 # find dependencies
 find_package(ament_cmake REQUIRED)
@@ -86,10 +90,16 @@ PUBLIC
   rclcpp
   sensor_msgs)
 
+if(MSVC)
+  find_package(Iconv REQUIRED)
+endif()
+
 target_link_libraries(${PROJECT_NAME}
   PUBLIC
     opencv_imgproc
-    PkgConfig::LIBAV)
+    PkgConfig::LIBAV
+    $<$<BOOL:${MSVC}>:${Iconv_LIBRARIES}>
+)
 
 
 ament_export_targets(${PROJECT_NAME}Targets HAS_LIBRARY_TARGET)
