
diff --git a/moveit_ros/moveit_servo/demos/servo_keyboard_input.cpp b/moveit_ros/moveit_servo/demos/servo_keyboard_input.cpp
index 8a6507c569..9c5d0805a6 100644
--- a/moveit_ros/moveit_servo/demos/servo_keyboard_input.cpp
+++ b/moveit_ros/moveit_servo/demos/servo_keyboard_input.cpp
@@ -45,8 +45,12 @@
 #include <rclcpp/rclcpp.hpp>
 #include <signal.h>
 #include <stdio.h>
+#ifndef WIN32
 #include <termios.h>
 #include <unistd.h>
+#else
+#include <conio.h>
+#endif
 
 // Define used keys
 namespace
@@ -88,6 +92,7 @@ class KeyboardReader
 public:
   KeyboardReader() : file_descriptor_(0)
   {
+#ifndef WIN32
     // get the console in raw mode
     tcgetattr(file_descriptor_, &cooked_);
     struct termios raw;
@@ -97,23 +102,32 @@ class KeyboardReader
     raw.c_cc[VEOL] = 1;
     raw.c_cc[VEOF] = 2;
     tcsetattr(file_descriptor_, TCSANOW, &raw);
+#endif
   }
   void readOne(char* c)
   {
+#ifndef WIN32
     int rc = read(file_descriptor_, c, 1);
     if (rc < 0)
     {
       throw std::runtime_error("read failed");
     }
+#else
+    *c = static_cast<char>(_getch());
+#endif
   }
   void shutdown()
   {
+#ifndef WIN32
     tcsetattr(file_descriptor_, TCSANOW, &cooked_);
+#endif
   }
 
 private:
   int file_descriptor_;
+#ifndef WIN32
   struct termios cooked_;
+#endif
 };
 
 // Converts key-presses to Twist or Jog commands for Servo, in lieu of a controller
