diff --git a/src/moveit/moveit_py_utils/CMakeLists.txt b/src/moveit/moveit_py_utils/CMakeLists.txt
index 132da7425..5e6564626 100644
--- a/src/moveit/moveit_py_utils/CMakeLists.txt
+++ b/src/moveit/moveit_py_utils/CMakeLists.txt
@@ -8,6 +8,7 @@ set_target_properties(moveit_py_utils PROPERTIES VERSION
 
 ament_target_dependencies(moveit_py_utils rclcpp moveit_msgs geometry_msgs
                           pybind11)
+target_link_libraries(moveit_py_utils Python3::Module)
 
 install(
   TARGETS moveit_py_utils
