diff --git a/include/rviz_common/tool_manager.hpp b/include/rviz_common/tool_manager.hpp
index cb1711706..a70622342 100644
--- a/include/rviz_common/tool_manager.hpp
+++ b/include/rviz_common/tool_manager.hpp
@@ -40,6 +40,7 @@

 #include "rviz_common/factory/pluginlib_factory.hpp"
 #include "rviz_common/tool.hpp"
+#include "rviz_common/visibility_control.hpp"

 class QKeyEvent;

@@ -53,7 +54,7 @@ class PropertyTreeModel;

 }  // namespace properties

-class ToolManager : public QObject
+class RVIZ_COMMON_PUBLIC ToolManager : public QObject
 {
   Q_OBJECT

