diff --git a/CMakeLists.txt b/CMakeLists.txt
index f6c0bec..ae46334 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -37,6 +37,22 @@ add_library(${PROJECT_NAME}
 )
 
 target_link_libraries(${PROJECT_NAME} ${DRACO_LIBRARIES})
+# draco 1.5.3 dropped the DRACO_LIBRARIES variable in favor of
+# draco::draco imported target, but debian introduced DRACO_LIBRARIES
+# back, but that is not present in other distributions. For maximum
+# compatibility, we link DRACO_LIBRARIES if defined, otherwise draco::draco
+# See:
+# * https://github.com/google/draco/commit/7109fbee87c6d932cdc0ac7394bc5d485db26d44
+# * https://sources.debian.org/patches/draco/1.5.6+dfsg-3/0004-Set-DRACO_LIBRARIES-for-backwards-compatibility.patch/
+if(DEFINED DRACO_LIBRARIES)
+  target_link_libraries(${PROJECT_NAME} PRIVATE
+    ${DRACO_LIBRARIES}
+  )
+else()
+  target_link_libraries(${PROJECT_NAME} PRIVATE
+    draco::draco
+  )
+endif()
 
 ament_target_dependencies(${PROJECT_NAME} ${dependencies})

