diff --git a/CMakeLists.txt b/CMakeLists.txt
index 29506fe4..6fbfc480 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,8 +8,8 @@ include(ExternalProject)
 ExternalProject_Add(upstr_lely_core_libraries    # Name for custom target
   #--Download step--------------
   SOURCE_DIR ${CMAKE_CURRENT_BINARY_DIR}/upstream
-  INSTALL_DIR "${CMAKE_INSTALL_PREFIX}"  # Installation prefix
-  BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/build
+  INSTALL_DIR "${CMAKE_CURRENT_BINARY_DIR}/lely_install"  # Installation prefix
+  BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/lely
   GIT_REPOSITORY https://gitlab.com/lely_industries/lely-core.git
   GIT_TAG fb735b79cab5f0cdda45bc5087414d405ef8f3ab
   TIMEOUT 60
@@ -21,13 +21,13 @@ ExternalProject_Add(upstr_lely_core_libraries    # Name for custom target
   CONFIGURE_COMMAND autoreconf -i <SOURCE_DIR>
   COMMAND <SOURCE_DIR>/configure --prefix=<INSTALL_DIR> --disable-cython --disable-doc --disable-tests --disable-static --disable-diag
   #BUILD STEP execute make
-  BUILD_COMMAND $(MAKE) -C ${CMAKE_CURRENT_BINARY_DIR}/build
+  BUILD_COMMAND make -j
   #INSTALL STEP do nothing as we install in main
-  INSTALL_COMMAND ""
+  INSTALL_COMMAND make install VERBOSE=1
 )
 
 #INSTALL lely_core_libraries - execute make install
-install(CODE "execute_process(COMMAND ${CMAKE_MAKE_PROGRAM} install VERBOSE=1 WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/build)")
+#install(CODE "execute_process(COMMAND ${CMAKE_MAKE_PROGRAM} install VERBOSE=1 WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/build)")
 
 set(lely_core_cmake_DIR "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 include("cmake/lely_core_libraries-extras.cmake" NO_POLICY_SCOPE)
@@ -46,6 +46,22 @@ install(
     DESTINATION ${CMAKE_INSTALL_PREFIX}/bin/
     USE_SOURCE_PERMISSIONS)
 
+install(
+    DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/lely_install/bin/
+    USE_SOURCE_PERMISSIONS
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/bin
+)
+
+install(
+    DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/lely_install/include/lely/
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/include/lely
+)
+
+install(
+    DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/lely_install/lib/
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
+)
+
 ament_export_include_directories(include)
 ament_export_libraries(lely-can lely-co lely-coapp lely-ev lely-io2 lely-libc lely-tap lely-util)
 ament_package(
