diff --git a/src/vendored/dynmsg/message_reading_cpp.cpp b/src/vendored/dynmsg/message_reading_cpp.cpp
index 16a3cd1..743008a 100644
--- a/src/vendored/dynmsg/message_reading_cpp.cpp
+++ b/src/vendored/dynmsg/message_reading_cpp.cpp
@@ -398,7 +398,7 @@ dynamic_array_to_yaml_impl_bool(
   static_cast<void>(member_info);
   for (size_t ii = 0; ii < v->size(); ++ii) {
     DYNMSG_DEBUG(std::cout << (v->operator[](ii) ? "true" : "false") << ", ");
-    array_node.push_back(v->operator[](ii));
+    array_node.push_back(static_cast<bool>(v->operator[](ii)));
   }
   DYNMSG_DEBUG(std::cout << std::endl);
 }
