diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -97,6 +97,18 @@ if (CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64")
 elseif(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
   set(FOXGLOVE_SDK_PLATFORM "x86_64-unknown-linux-gnu")
   set(FOXGLOVE_SDK_SHA "8fb03b061127788ad708918d186d74d4fdb58718b11cff97c9d605513ab57cc6")
+elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64")
+  set(FOXGLOVE_SDK_PLATFORM "aarch64-apple-darwin")
+  set(FOXGLOVE_SDK_SHA "aae2ccfebcce4b6ff72e366912a92d6576f6100e79cdf71d79e4288977add0f0")
+elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
+  set(FOXGLOVE_SDK_PLATFORM "x86_64-apple-darwin")
+  set(FOXGLOVE_SDK_SHA "4e592b4fe447b9fab912706f863a75833cf6bedc5c17d3ea474d4531eec722ae")
+elseif(CMAKE_SYSTEM_NAME STREQUAL "Windows" AND CMAKE_SYSTEM_PROCESSOR STREQUAL "ARM64")
+  set(FOXGLOVE_SDK_PLATFORM "aarch64-pc-windows-msvc")
+  set(FOXGLOVE_SDK_SHA "aaafd94b67ce9de06eea616f247e49c0b11a2f78cf26b7cb1e1ae45f264a4067")
+elseif(CMAKE_SYSTEM_NAME STREQUAL "Windows" AND CMAKE_SYSTEM_PROCESSOR STREQUAL "AMD64")
+  set(FOXGLOVE_SDK_PLATFORM "x86_64-pc-windows-msvc")
+  set(FOXGLOVE_SDK_SHA "335fcaa4d2637a529d6a47739feb944defcd393b9ccdb8a2fe2901747dbff863")
 else()
   message(FATAL_ERROR "Unsupported platform/architecture combination: ${CMAKE_SYSTEM_PROCESSOR}-${CMAKE_SYSTEM_NAME}")
 endif()
diff --git a/src/message_definition_cache.cpp b/src/message_definition_cache.cpp
--- a/src/message_definition_cache.cpp
+++ b/src/message_definition_cache.cpp
@@ -3,6 +3,7 @@
 #include <filesystem>
 #include <fstream>
 #include <functional>
+#include <map>
 #include <optional>
 #include <regex>
 #include <set>
