From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: RoboStack <maintainers@robostack.org>
Date: Wed, 13 May 2026 13:10:00 +0000
Subject: [PATCH] Relax urdfdom_headers version pin

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,6 +15,6 @@ find_package(ament_cmake_ros REQUIRED)
 find_package(pluginlib REQUIRED)
 find_package(rcutils REQUIRED)
-find_package(urdfdom_headers 1.0.6 REQUIRED)
+find_package(urdfdom_headers REQUIRED)
 find_package(urdf_parser_plugin REQUIRED)
 find_package(tinyxml2_vendor REQUIRED)
 find_package(TinyXML2 REQUIRED)
-- 
2.43.0
