diff --git a/include/mavconn/io_context_runner.hpp b/include/mavconn/io_context_runner.hpp
index 9db0b10..789f3b5 100644
--- a/include/mavconn/io_context_runner.hpp
+++ b/include/mavconn/io_context_runner.hpp
@@ -59,7 +59,7 @@ public:
       return;
     }
 
-    io_thread_ = std::jthread(
+    io_thread_ = std::thread(
       [this, f = std::forward<Fn>(fn)]() mutable {
         is_running_ = true;
         f();
@@ -73,7 +73,6 @@ public:
       return;
     }
 
-    io_thread_.request_stop();
     io_work_.reset();
     io_.stop();
   }
@@ -119,7 +118,7 @@ private:
   std::unique_ptr<asio::executor_work_guard<asio::io_context::executor_type>> io_work_;
   bool owns_thread_;
   std::atomic<bool> is_running_;
-  std::jthread io_thread_;
+  std::thread io_thread_;
 };
 
 }  // namespace mavconn
