diff --git a/src/zstd_publisher.cpp b/src/zstd_publisher.cpp
index b215930..301a58c 100644
--- a/src/zstd_publisher.cpp
+++ b/src/zstd_publisher.cpp
@@ -86,4 +86,4 @@ void ZstdPublisher::advertiseImpl(
-  uint ns_len = node->get_effective_namespace().length();
+  unsigned int ns_len = node->get_effective_namespace().length();
   std::string param_base_name = base_topic.substr(ns_len);
   std::replace(param_base_name.begin(), param_base_name.end(), '/', '.');
 
