diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2e80ed198f..1b633d0a2f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,6 +48,7 @@ add_library(
   src/steering_controllers_library.cpp
   src/steering_odometry.cpp
 )
+target_compile_definitions(steering_controllers_library PRIVATE _USE_MATH_DEFINES)
 target_compile_features(steering_controllers_library PUBLIC cxx_std_17)
 target_include_directories(steering_controllers_library PUBLIC
   "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"
