diff --git a/include/usb_cam/formats/av_pixel_format_helper.hpp b/include/usb_cam/formats/av_pixel_format_helper.hpp
index ad1792ec..a8fa2328 100644
--- a/include/usb_cam/formats/av_pixel_format_helper.hpp
+++ b/include/usb_cam/formats/av_pixel_format_helper.hpp
@@ -743,12 +743,6 @@ const std::unordered_map<std::string, AVPixelFormat> STR_2_AVPIXFMT = {
                                                                       ///< 16-bit samples,
                                                                       ///< big-endian
 
-
-  {stringify(AV_PIX_FMT_XVMC), AV_PIX_FMT_XVMC},                      ///< XVideo Motion
-                                                                      ///< Acceleration via common
-                                                                      ///< packet passing
-
-
   {stringify(AV_PIX_FMT_YUV440P10LE), AV_PIX_FMT_YUV440P10LE},        ///< planar YUV 4:4:0,20bpp,
                                                                       ///< (1 Cr & Cb sample per
                                                                       ///< 1x2 Y samples),
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1a83c1e..fcfb823 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,7 +7,7 @@ if(NOT CMAKE_CXX_STANDARD)
 endif()
 
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-  add_compile_options(-Wall -Wextra -Wpedantic -Werror)
+  add_compile_options(-Wall -Wextra -Wpedantic)
 endif()
 
 find_package(ament_cmake_auto REQUIRED)