diff -ruN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -477,7 +477,7 @@
 # Find Eigen3. If it defines the target, this is used. If not,
 # fall back to the using the module form.
 # See https://eigen.tuxfamily.org/dox/TopicCMakeGuide.html for details
-find_package(Eigen3 3.3 REQUIRED)
+find_package(Eigen3 REQUIRED NO_MODULE)
 if (TARGET Eigen3::Eigen)
   set(G2O_EIGEN3_EIGEN_TARGET Eigen3::Eigen)
 else()
diff -ruN a/g2o/examples/sphere/create_sphere.cpp b/g2o/examples/sphere/create_sphere.cpp
--- a/g2o/examples/sphere/create_sphere.cpp
+++ b/g2o/examples/sphere/create_sphere.cpp
@@ -166,8 +166,8 @@
     cerr << "using seeds:";
     for (size_t i = 0; i < seeds.size(); ++i) cerr << " " << seeds[i];
     cerr << endl;
-    transSampler.seed(seeds[0]);
-    rotSampler.seed(seeds[1]);
+    transSampler.seed(static_cast<int>(seeds[0]));
+    rotSampler.seed(static_cast<int>(seeds[1]));
   }
 
   // noise for all the edges
diff -ruN a/g2o/stuff/misc.h b/g2o/stuff/misc.h
--- a/g2o/stuff/misc.h
+++ b/g2o/stuff/misc.h
@@ -27,6 +27,7 @@
 #ifndef G2O_STUFF_MISC_H
 #define G2O_STUFF_MISC_H
 
+#include <cassert>
 #include <cmath>
 
 /** @addtogroup utils **/
