diff --git a/draco_point_cloud_transport/CMakeLists.txt b/draco_point_cloud_transport/CMakeLists.txt
index a2c6e27..b88f5f8 100644
--- a/draco_point_cloud_transport/CMakeLists.txt
+++ b/draco_point_cloud_transport/CMakeLists.txt
@@ -36,9 +36,26 @@ add_library(${PROJECT_NAME}
 )

 target_link_libraries(${PROJECT_NAME} PRIVATE
-  ${DRACO_LIBRARIES}
   ${dependencies}
 )
+
+# draco 1.5.3 dropped the DRACO_LIBRARIES variable in favor of
+# draco::draco imported target, but debian introduced DRACO_LIBRARIES
+# back, but that is not present in other distributions. For maximum
+# compatibility, we link DRACO_LIBRARIES if defined, otherwise draco::draco
+# See:
+# * https://github.com/google/draco/commit/7109fbee87c6d932cdc0ac7394bc5d485db26d44
+# * https://sources.debian.org/patches/draco/1.5.6+dfsg-3/0004-Set-DRACO_LIBRARIES-for-backwards-compatibility.patch/
+if(DEFINED DRACO_LIBRARIES)
+  target_link_libraries(${PROJECT_NAME} PRIVATE
+    ${DRACO_LIBRARIES}
+  )
+else()
+  target_link_libraries(${PROJECT_NAME} PRIVATE
+    draco::draco
+  )
+endif()
+
 target_include_directories(${PROJECT_NAME} PRIVATE
   "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
   "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>"

diff --git a/point_cloud_transport/include/point_cloud_transport/simple_publisher_plugin.hpp b/point_cloud_transport/include/point_cloud_transport/simple_publisher_plugin.hpp
index 394fde1..39ab353 100644
--- a/point_cloud_transport/include/point_cloud_transport/simple_publisher_plugin.hpp
+++ b/point_cloud_transport/include/point_cloud_transport/simple_publisher_plugin.hpp
@@ -95,7 +95,7 @@ class SimplePublisherPlugin : public point_cloud_transport::PublisherPlugin
   bool getParam(const std::string & parameter_name, T & value) const
   {
     if (simple_impl_) {
-      uint ns_len = strlen(simple_impl_->node_interfaces_
+      unsigned int ns_len = strlen(simple_impl_->node_interfaces_
           .get_node_base_interface()->get_namespace());
       std::string param_base_name = getTopic().substr(ns_len);
       std::replace(param_base_name.begin(), param_base_name.end(), '/', '.');
@@ -120,7 +120,7 @@ class SimplePublisherPlugin : public point_cloud_transport::PublisherPlugin
   {
     if (simple_impl_) {
       // Declare Parameters
-      uint ns_len = strlen(simple_impl_->node_interfaces_
+      unsigned int ns_len = strlen(simple_impl_->node_interfaces_
           .get_node_base_interface()->get_namespace());
       std::string param_base_name = getTopic().substr(ns_len);
       std::replace(param_base_name.begin(), param_base_name.end(), '/', '.');
