diff --git a/src/crop_foremost.cpp b/src/crop_foremost.cpp
index 60901d9..6535c72 100644
--- a/src/crop_foremost.cpp
+++ b/src/crop_foremost.cpp
@@ -135,7 +135,7 @@ void CropForemostNode::depthCb(const sensor_msgs::msg::Image::ConstSharedPtr & r
     case CV_8UC1:
     case CV_8SC1:
     case CV_32F:
-      cv::threshold(cv_ptr->image, cv_ptr->image, minVal + distance_, 0, CV_THRESH_TOZERO_INV);
+      cv::threshold(cv_ptr->image, cv_ptr->image, minVal + distance_, 0, cv::THRESH_TOZERO_INV);
       break;
     case CV_16UC1:
     case CV_16SC1:
@@ -143,7 +143,7 @@ void CropForemostNode::depthCb(const sensor_msgs::msg::Image::ConstSharedPtr & r
     case CV_64F:
       // 8 bit or 32 bit floating array is required to use cv::threshold
       cv_ptr->image.convertTo(cv_ptr->image, CV_32F);
-      cv::threshold(cv_ptr->image, cv_ptr->image, minVal + distance_, 1, CV_THRESH_TOZERO_INV);
+      cv::threshold(cv_ptr->image, cv_ptr->image, minVal + distance_, 1, cv::THRESH_TOZERO_INV);
 
       cv_ptr->image.convertTo(cv_ptr->image, imtype);
       break;
