diff --git a/CMakeLists.txt b/CMakeLists.txt
index e07dcf4..1b430f8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -78,7 +78,7 @@ target_link_libraries(ouster_ros_library
     ouster_build
     pcl_common
   # PRIVATE (unsupported)
-  -Wl,--whole-archive ouster_client -Wl,--no-whole-archive
+    ouster_client
 )
 
 # helper method to construct ouster-ros components
diff --git a/src/os_ros.cpp b/src/os_ros.cpp
index 0ddbf8e..a326b1a 100644
--- a/src/os_ros.cpp
+++ b/src/os_ros.cpp
@@ -127,7 +127,7 @@ void copy_scan_to_cloud(ouster_ros::Cloud& cloud, const ouster::LidarScan& ls,
     const auto rf = reflectivity.data();
     const auto nr = near_ir.data();
     const auto sg = signal.data();
-    const auto t_zero = std::chrono::duration<long int, std::nano>::zero();
+    constexpr std::chrono::nanoseconds t_zero{0};
 
 #ifdef __OUSTER_UTILIZE_OPENMP__
 #pragma omp parallel for collapse(2)
