diff --git a/CMakeLists.txt b/CMakeLists.txt
index 71deedd..c470d9a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -87,9 +87,10 @@ target_sources(foxglove_cpp_shared PRIVATE ${FOXGLOVE_SDK_SOURCES})
 # Import the prebuilt C shared library from the fetched SDK
 add_library(foxglove-shared SHARED IMPORTED)
 set_target_properties(foxglove-shared PROPERTIES
-  IMPORTED_LOCATION ${foxglove_sdk_SOURCE_DIR}/lib/libfoxglove.so
+  IMPORTED_LOCATION ${foxglove_sdk_SOURCE_DIR}/lib/foxglove.dll
+  IMPORTED_IMPLIB ${foxglove_sdk_SOURCE_DIR}/lib/foxglove.dll.lib
 )
-target_link_libraries(foxglove_cpp_shared PRIVATE foxglove-shared)
+target_link_libraries(foxglove_cpp_shared PRIVATE foxglove-shared Ws2_32 Userenv Ntdll)
 set_target_properties(foxglove_cpp_shared PROPERTIES
   INSTALL_RPATH "$ORIGIN"
   BUILD_WITH_INSTALL_RPATH TRUE
