diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6b642f8..31cedab 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,10 +1,20 @@
 cmake_minimum_required(VERSION 3.10)
 project(gz_dartsim_vendor)
 
+# Project-specific settings
+set(LIB_VER_MAJOR 6)
+set(LIB_VER_MINOR 13)
+set(LIB_VER_PATCH 2)
+set(LIB_VER ${LIB_VER_MAJOR}.${LIB_VER_MINOR}.${LIB_VER_PATCH})
+
 find_package(ament_cmake REQUIRED)
 find_package(ament_cmake_vendor_package REQUIRED)
 
+find_package(DART ${LIB_VER})
+message(NOTICE "DART_FOUND" = ${DART_FOUND})
+
 ament_vendor(${PROJECT_NAME}
+  SATISFIED ${DART_FOUND}
   VCS_URL https://github.com/dartsim/dart.git
   VCS_VERSION v6.13.2
   CMAKE_ARGS
