diff --git a/include/point_cloud_transport/simple_publisher_plugin.hpp b/include/point_cloud_transport/simple_publisher_plugin.hpp
index 38d2ad1..2ddbcdb 100644
--- a/include/point_cloud_transport/simple_publisher_plugin.hpp
+++ b/include/point_cloud_transport/simple_publisher_plugin.hpp
@@ -94,7 +94,7 @@ public:
   bool getParam(const std::string & parameter_name, T & value) const
   {
     if (simple_impl_) {
-      uint ns_len = simple_impl_->node_->get_effective_namespace().length();
+      unsigned int ns_len = simple_impl_->node_->get_effective_namespace().length();
       std::string param_base_name = getTopic().substr(ns_len);
       std::replace(param_base_name.begin(), param_base_name.end(), '/', '.');
 
@@ -113,7 +113,7 @@ public:
   {
     if (simple_impl_) {
       // Declare Parameters
-      uint ns_len = simple_impl_->node_->get_effective_namespace().length();
+      unsigned int ns_len = simple_impl_->node_->get_effective_namespace().length();
       std::string param_base_name = getTopic().substr(ns_len);
       std::replace(param_base_name.begin(), param_base_name.end(), '/', '.');
 
diff --git a/include/point_cloud_transport/simple_subscriber_plugin.hpp b/include/point_cloud_transport/simple_subscriber_plugin.hpp
index a4958ff..1a2fb4d 100644
--- a/include/point_cloud_transport/simple_subscriber_plugin.hpp
+++ b/include/point_cloud_transport/simple_subscriber_plugin.hpp
@@ -92,7 +92,7 @@ public:
   bool getParam(const std::string & parameter_name, T & value) const
   {
     if (impl_) {
-      uint ns_len = impl_->node_->get_effective_namespace().length();
+      unsigned int ns_len = impl_->node_->get_effective_namespace().length();
       std::string param_base_name = getTopic().substr(ns_len);
       std::replace(param_base_name.begin(), param_base_name.end(), '/', '.');
 
@@ -110,7 +110,7 @@ public:
     rcl_interfaces::msg::ParameterDescriptor())
   {
     if (impl_) {
-      uint ns_len = impl_->node_->get_effective_namespace().length();
+      unsigned int ns_len = impl_->node_->get_effective_namespace().length();
       std::string param_base_name = getTopic().substr(ns_len);
       std::replace(param_base_name.begin(), param_base_name.end(), '/', '.');
 
