diff --git a/src/collisions_updater.cpp b/src/collisions_updater.cpp
index d8963e1be..f99e493ac 100644
--- a/src/collisions_updater.cpp
+++ b/src/collisions_updater.cpp
@@ -100,7 +100,7 @@ int main(int argc, char* argv[])
     auto package_settings = config_data->get<moveit_setup::PackageSettingsConfig>("package_settings");
     try
     {
-      package_settings->loadExisting(config_pkg_path);
+      package_settings->loadExisting(config_pkg_path.string());
     }
     catch (const std::runtime_error& e)
     {
@@ -113,7 +113,7 @@ int main(int argc, char* argv[])
     RCLCPP_ERROR_STREAM(node->get_logger(), "Please provide config package or URDF and SRDF path");
     return 1;
   }
-  else if (rdf_loader::RDFLoader::isXacroFile(srdf_path) && output_path.empty())
+  else if (rdf_loader::RDFLoader::isXacroFile(srdf_path.string()) && output_path.empty())
   {
     RCLCPP_ERROR_STREAM(node->get_logger(), "Please provide a different output file for SRDF xacro input file");
     return 1;
diff --git a/src/setup_assistant_widget.cpp b/src/setup_assistant_widget.cpp
index 2d56b5e40..bab1ec094 100644
--- a/src/setup_assistant_widget.cpp
+++ b/src/setup_assistant_widget.cpp
@@ -71,7 +71,7 @@ SetupAssistantWidget::SetupAssistantWidget(const rviz_common::ros_integration::R
 
   // Setting the window icon
   auto icon_path = getSharePath("moveit_ros_visualization") / "icons/classes/MotionPlanning.png";
-  setWindowIcon(QIcon(icon_path.c_str()));
+  setWindowIcon(QIcon(icon_path.string().c_str()));
 
   // Basic widget container -----------------------------------------
   QHBoxLayout* layout = new QHBoxLayout();
