diff --git a/CMakeLists.txt b/CMakeLists.txt
index 99a8ccc..b59e949 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,6 @@
 # judge which cmake codes to use 
+set(ROS_EDITION "ROS2")
+set(HUMBLE_ROS "humble")
 if(ROS_EDITION STREQUAL "ROS1")
 
   # Copyright(c) 2019 livoxtech limited.
@@ -190,6 +192,7 @@ else(ROS_EDITION STREQUAL "ROS2")
 
   cmake_minimum_required(VERSION 3.14)
   project(livox_ros_driver2)
+  configure_file(${CMAKE_CURRENT_SOURCE_DIR}/package_ROS2.xml ${CMAKE_CURRENT_SOURCE_DIR}/package.xml)
 
   # Default to C99
   if(NOT CMAKE_C_STANDARD)

