diff --git a/CMakeLists.txt b/CMakeLists.txt
index e8420aec3..b1257cde3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -64,7 +64,7 @@ generate_parameter_library(
 )
 
 add_library(planning_context_loader_base SHARED src/planning_context_loader.cpp)
-target_link_libraries(planning_context_loader_base cartesian_limits_parameters)
+target_link_libraries(planning_context_loader_base cartesian_limits_parameters joint_limits_common)
 ament_target_dependencies(planning_context_loader_base
                           ${THIS_PACKAGE_INCLUDE_DEPENDS})
 
@@ -79,7 +79,7 @@ add_library(
   trajectory_generation_common SHARED
   src/trajectory_functions.cpp src/trajectory_generator.cpp
   src/trajectory_blender_transition_window.cpp)
-target_link_libraries(trajectory_generation_common cartesian_limits_parameters)
+target_link_libraries(trajectory_generation_common cartesian_limits_parameters joint_limits_common)
 ament_target_dependencies(trajectory_generation_common
                           ${THIS_PACKAGE_INCLUDE_DEPENDS})
 
