diff --git a/ompl_interface/CMakeLists.txt b/ompl_interface/CMakeLists.txt
index e22c868ae4..c7647bcb38 100644
--- a/ompl_interface/CMakeLists.txt
+++ b/ompl_interface/CMakeLists.txt
@@ -22,7 +22,12 @@ add_library(
 set_target_properties(moveit_ompl_interface
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
 
-find_package(OpenMP REQUIRED)
+if(APPLE)
+  find_package(OpenMP)
+  set(OpenMP_CXX_FLAGS "-fopenmp")
+else()
+  find_package(OpenMP REQUIRED)
+endif()
 
 ament_target_dependencies(
   moveit_ompl_interface
