diff --git a/ament_clang_tidy/ament_clang_tidy/main.py b/ament_clang_tidy/ament_clang_tidy/main.py
index 5faa81425..e8c8c12c3 100755
--- a/ament_clang_tidy/ament_clang_tidy/main.py
+++ b/ament_clang_tidy/ament_clang_tidy/main.py
@@ -30,6 +30,7 @@
 
 import yaml
 
+from ament_lint.filesystem_helpers import find_executable
 
 def main(argv=sys.argv[1:]):
     extensions = ['c', 'cc', 'cpp', 'cxx', 'h', 'hh', 'hpp', 'hxx']
@@ -265,16 +266,6 @@ def start_subprocess(full_cmd):
             f.write(xml)
 
 
-def find_executable(file_names):
-    paths = os.getenv('PATH').split(os.path.pathsep)
-    for file_name in file_names:
-        for path in paths:
-            file_path = os.path.join(path, file_name)
-            if os.path.isfile(file_path) and os.access(file_path, os.X_OK):
-                return file_path
-    return None
-
-
 def get_compilation_db_files(paths):
     files = []
     for path in paths: