diff --git a/CMakeLists.txt b/CMakeLists.txt
index c5f5c788a..dbb0f0d3b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,7 +7,7 @@ if(NOT CMAKE_CXX_STANDARD)
 endif()
 
 if(CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-  add_compile_options(-Wall -Wextra -Wpedantic -Werror)
+  add_compile_options(-Wall -Wextra -Wpedantic)
 endif()
 
 # Qt5 boilerplate options from http://doc.qt.io/qt-5/cmake-manual.html
diff --git a/src/costmap_cost_tool.cpp b/src/costmap_cost_tool.cpp
index f25a71cd8..93d234043 100644
--- a/src/costmap_cost_tool.cpp
+++ b/src/costmap_cost_tool.cpp
@@ -50,6 +50,10 @@ void CostmapCostTool::onInitialize()
     node_->create_client<nav2_msgs::srv::GetCost>("/global_costmap/get_cost_global_costmap");
 }
 
+void CostmapCostTool::updateAutoDeactivate() {
+    // TODO: Add implementation
+}
+
 void CostmapCostTool::activate() {}
 void CostmapCostTool::deactivate() {}
 
diff --git a/src/docking_panel.cpp b/src/docking_panel.cpp
index 62304a825..291a570f9 100644
--- a/src/docking_panel.cpp
+++ b/src/docking_panel.cpp
@@ -311,6 +311,10 @@ void DockingPanel::save(rviz_common::Config config) const
   Panel::save(config);
 }
 
+void DockingPanel::onStartup() {
+    // TODO: Add implementation
+}
+
 void DockingPanel::onDockingButtonPressed()
 {
   auto is_action_server_ready =
